bootloader/bootloader.go

```diff
@@ -352,8 +352,17 @@ func ForceError(err error) {
 	forcedError = err
 }
 
-func extractKernelAssetsToBootDir(dstDir string, snapf snap.Container, assets []string) error {
-	// now do the kernel specific bits
+// extractKernelAssetsToBootDir will extract the given assets into the
+// given assetsDir. If the target directory already exists it will do
+// nothing.
+func extractKernelAssetsToBootDir(assetsDir string, snapf snap.Container, assets []string) error {
+	// Final dir already in place: nothing more to do.
+	if st, err := os.Stat(assetsDir); err == nil && st.IsDir() {
```

Comment:
nitpick, but we have a helper for this
```suggestion
	if exists, isDir, _ := osutil.DirExists(); exists && isDir {
```